management/virtualmachinedisk/client.go

```diff
@@ -17,15 +27,207 @@ func NewClient(client management.Client) DiskClient {
 	return DiskClient{client: client}
 }
 
-func (c DiskClient) DeleteDisk(diskName string, deleteVhdToo bool) (management.OperationID, error) {
-	if diskName == "" {
-		return "", fmt.Errorf(errParamNotSpecified, "diskName")
+// AddDataDisk adds a data disk to a Virtual Machine
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157199.aspx
+func (c DiskClient) AddDataDisk(
+	service string,
+	deployment string,
+	role string,
+	params CreateDataDiskParameters) (management.OperationID, error) {
+	if service == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "service")
+	}
+	if deployment == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "deployment")
+	}
+	if role == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "role")
+	}
+
+	requestURL := fmt.Sprintf(addDataDiskURL, service, deployment, role)
+
+	req, err := xml.Marshal(params)
+	if err != nil {
+		return "", err
+	}
+
+	return c.client.SendAzurePostRequest(requestURL, req)
+}
+
+// AddDisk adds an operating system disk or data disk to the user image repository
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157178.aspx
+func (c DiskClient) AddDisk(params CreateDiskParameters) (management.OperationID, error) {
+	requestURL := fmt.Sprintf(addDiskURL)
```

Comment:
Go ahead and update. I'll give Ahmet a bit more time to cast a veto before merging anyway...